### DISCUSSION TITLE
Why is Memtest86+ testing memory beyond the end?

### DISCUSSION THREAD

**Discussion by MacGyverNL** [memtest86plus/memtest86plus#193] [Q&A] [answered]
CPU: Intel i9 10940x 14 cores, 28 threads
Mainboard: ASRock x299 Taichi CLX
Memory: 32GiB Corsair Vengeance LPX 2400MHz

The memory on this system as reported by memtest86+ version 6.00 is 31.6 GB. But when it runs any actual tests, it tests ranges up to and including 34GB (without errors). Is this simply a mismatch between GiB and GB, or is there something more sinister going on?

If it's a mismatch between GiB and GB, perhaps this could be clarified in the interface -- it is *very* weird to see 32-33GB being tested on a system with only 31.6GB memory reported.

**Comment by debrouxl** [comment, marked as answer]
0-34 GB is extremely likely to be just 32 GB + 2 GB of memory holes. See also https://github.com/memtest86plus/memtest86plus/issues/36#issuecomment-1094106263 , where I mentioned one of my computers which has a 1.5 GB memory hole when one of my old graphics cards is plugged in (only 0.5 GB with a different graphics card), or the FAQ entry on https://www.memtest.org :

> > Why is Memtest86+ testing more memory than I have?

> It doesn't. The memory on a modern computer is not necessarily mapped in a linear way. E.g.: 16GB of RAM can be virtually mapped from 0-15GB and 16-17GB with a hole between 15GB and 16GB.

**Reply by Infinoid** [reply]
> 0-34 GB is extremely likely to be just 32 GB + 2 GB of memory holes.

That's fine.  But it seems like it will be a constant source of confusion.  Can the UI be adjusted to talk about which *memory* is being tested, instead of which *address* it is testing?  This would make things much clearer to users.

**Comment by MacGyverNL** [comment]
> 0-34 GB is extremely likely to be just 32 GB + 2 GB of memory holes. See also [#36 (comment)](https://github.com/memtest86plus/memtest86plus/issues/36#issuecomment-1094106263) , where I mentioned one of my computers which has a 1.5 GB memory hole when one of my old graphics cards is plugged in (only 0.5 GB with a different graphics card),

Oh that's likely it then, this machine has two GPUs plugged in. I didn't find that issue with github search, probably because it's too broad for the search terms I used to hit it.

> or the FAQ entry on https://www.memtest.org :
> 
> > > Why is Memtest86+ testing more memory than I have?
> 
> > It doesn't. The memory on a modern computer is not necessarily mapped in a linear way. E.g.: 16GB of RAM can be virtually mapped from 0-15GB and 16-17GB with a hole between 15GB and 16GB.

Sorry, I completely missed this, I immediately went searching github issues instead of checking the FAQ :')
